src/Compilers/Core/Portable/CommandLine/SarifVersion.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Specifies the version of the SARIF log file to produce.
+    /// </summary>
+    public enum SarifVersion
+    {
+        /// <summary>
+        /// The original, non-standardized version of the SARIF format.
+        /// </summary>
+        Sarif1 = 1,
+
+        /// <summary>
+        /// The first standardized version of the SARIF format.
+        /// </summary>
+        Sarif2 = 2,
+
+        /// <summary>
+        /// The default SARIF version, which is v1.0.0 for compatibility with
+        /// previous versions of the compiler.
+        /// </summary>
+        Default = Sarif1,
+
+        /// <summary>
+        /// The latest supported SARIF version.
+        /// </summary>
+        Latest = int.MaxValue
+    }
+
+    public static class SarifVersionFacts
+    {
+        /// <summary>
+        /// Try to parse the SARIF log file version from a string.
+        /// </summary>
+        public static bool TryParse(string version, out SarifVersion result)
+        {
+            if (version == null)
+            {
+                result = SarifVersion.Default;
+                return true;
+            }
+
+            switch (CaseInsensitiveComparison.ToLower(version))
```

Comment:
Why is `CaseInsensitiveComparison`, which performs a comparison for identifiers, better than an ordinal case-insensitive comparison here?

Reply:
Again, I modeled this code on `LanguageVersionFacts.TryParse`. Let me know what you and @jaredpar@microsoft.com decide.

---
In reply to: [322453003](https://github.com/dotnet/roslyn/pull/37148#discussion_r322453003) [](ancestors = 322453003)

Reply:
This is a one time only evaluation, so I'd go with `string.Equals(..., ..., StringComparison.OrdinalIgnoreCase)`.

For repeated executions, I usually go with something like:

```
private static readonly Dictionary<string, SarifVersion> versions = new Dictionary<string, SarifVersion>(StringComparer.OrdinalIgnoreCase)
    {
        { "default", SarifVersion.Default },
        { "latest", SarifVersion.Latest },
        { "1", SarifVersion.Sarif1 },
        { "1.0", SarifVersion.Sarif1 },
        { "1.0.0", SarifVersion.Sarif1 },
        { "2", SarifVersion.Sarif2 },
        { "2.1", SarifVersion.Sarif2 },
        { "2.1.0", SarifVersion.Sarif2 },
    };

public static bool TryParse(string version, out SarifVersion result)
{
    if (version is null)
    {
        result = SarifVersion.Default;
        return false;
    }
    
    return versions.TryGetValue(version, out result);
}
```

Reply:
Are you suggesting this?
```
if (string.Equals("default", version, StringComparison.OrdinalIgnoreCase) { return SarifVersion.Default; }
if (string.Equals("latest", version, StringComparison.OrdinalIgnoreCase) { return SarifVersion.Latest; }
if (string.Equals("1.0", version StringComparison.OrdinalIgnoreCase) { return SarifVersion.Sarif1; }
...
```
If so, why do you prefer that to the `switch` statement? Is it because the code is more compact? As you say, for a one-time parse of the command line, the perf difference (if any) is negligible.

---
In reply to: [323116124](https://github.com/dotnet/roslyn/pull/37148#discussion_r323116124) [](ancestors = 323116124)

Reply:
Allocations!